contracts/Controller.sol

```diff
@@ -614,13 +615,29 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
     /**
      * @notice get Otoken payout after expiry
      * @param _otoken Otoken address
-     * @param _amount amount of Otoken
-     * @return payout = cashValue * amount
+     * @param _amount amount of Otoken. Always in 1e18
+     * @return amount of collateral to payout
      */
     function _getPayout(address _otoken, uint256 _amount) internal view returns (uint256) {
         uint256 cashValue = calculator.getExpiredCashValue(_otoken);
+        uint256 amountInStrike = cashValue.mul(_amount).div(1e18);
 
-        return cashValue.mul(_amount).div(1e18);
+        OtokenInterface otoken = OtokenInterface(_otoken);
+        address strike = otoken.strikeAsset();
+        address collateral = otoken.collateralAsset();
+
+        uint256 payoutAmount;
+
+        if (strike == collateral) {
+            payoutAmount = amountInStrike;
+        } else {
+            uint256 expiry = otoken.expiryTimestamp();
+            (uint256 strikePrice, ) = oracle.getExpiryPrice(strike, expiry);
+            (uint256 collateralPrice, ) = oracle.getExpiryPrice(collateral, expiry);
+            payoutAmount = amountInStrike.mul(strikePrice).div(collateralPrice);
+        }
+
+        return _internalAmountToTokenAmount(payoutAmount, collateral);
```

Comment:
wait alternatively, can we have this calculation in calculator? Then we don't need oMath at all right?

Reply:
we can't have it in the getExpiredCashValue function, but we can use a separate function for that.